test/vsphere/e2e.sh

```diff
@@ -0,0 +1,172 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit  # exits immediately on any unexpected error (does not bypass traps)
+set -o nounset  # will error if variables are used without first being defined
+set -o pipefail # any non-zero exit code in a piped command causes the pipeline to fail with that code
+
+ARTIFACTS=$PWD/artifacts
+CERTS=$PWD/test/vsphere/openvpn/certs
+
+# Binary
+GO_VERSION=${GO_VERSION:-1.12}
+SOPS_VERSION=${SOPS_VERSION:-3.5.0}
+
+install_platformcli() {
+  docker run --rm -it -v "$PWD":"$PWD" -v /go:/go \
+    -w "$PWD" --entrypoint make -e GOPROXY=https://proxy.golang.org \
+    golang:"$GO_VERSION" pack build
+
+  sudo mv "$PWD"/.bin/platform-cli /usr/local/bin/platform-cli
+  sudo chmod +x /usr/local/bin/platform-cli
+  sudo apt-get update
+  sudo apt-get install -y genisoimage
+}
+
+install_kubectl() {
+  curl -L https://storage.googleapis.com/kubernetes-release/release/v1.17.0/bin/linux/amd64/kubectl -o /tmp/kubectl
+  sudo mv /tmp/kubectl /usr/local/bin/kubectl
+  sudo chmod +x /usr/local/bin/kubectl
+}
+
+install_sops() {
+  curl -L https://github.com/mozilla/sops/releases/download/v"$SOPS_VERSION"/sops-v"$SOPS_VERSION".linux -o /tmp/sops
+  sudo mv /tmp/sops /usr/local/bin/sops
+  sudo chmod +x /usr/local/bin/sops
+}
+
+decrypt_vpn_certs() {
+  export AWS_ACCESS_KEY_ID=$SOPS_AWS_ACCESS_KEY_ID
+  export AWS_SECRET_ACCESS_KEY=$SOPS_AWS_SECRET_ACCESS_KEY
+
+  for filename in "$CERTS"/*.enc; do
+    # save the decrypted certs in the same folder w/o the ext .enc
+    sops --config "$PWD"/test/vsphere/.sops.yaml -d \
+      --input-type binary --output-type binary "$filename" >"${filename%.*}"
+  done
+
+  # clean a WARNING in openvpn
+  # WARNING: file '/openvpn/certs/user.key' is group or others accessible
+  chmod 400 "$CERTS"/user.key
+}
+
+dump_logs() {
+  mkdir -p "$ARTIFACTS"/snapshot
+  platform-cli snapshot --output-dir "$ARTIFACTS"/snapshot
+}
+
+on_exit() {
+  # dump the logs into the ARTIFACTS directory
+  dump_logs
+
+  # remove the cluster
+  # shellcheck disable=SC2086
+  platform-cli cleanup $PLATFORM_OPTIONS_FLAGS
+
+  # kill the VPN
+  docker kill vpn
+
+  # clean certs dir - remove all except enc
+  rm "$CERTS"/{*.crt,*.key}
+}
+
+wait_for_vpn() {
+  while [ -n "$(ip addr show tun0 2>&1 >/dev/null)" ]; do
+    sleep 0.1
+  done
+}
+
+generate_cluster_id() {
+  local prefix
+
+  prefix=$(tr </dev/urandom -cd 'a-f0-9' | head -c 5)
+  echo "e2e-${prefix}"
+}
+
+# These are used inside e2e.yml
+PLATFORM_CLUSTER_ID=$(generate_cluster_id)
+export PLATFORM_CLUSTER_ID
+export PLATFORM_CA=$PWD/.certs/ca.crt.pem
+export PLATFORM_PRIVATE_KEY=$PWD/.certs/ca.key.pem
+export PLATFORM_CA_CEK=foobar
+export PLATFORM_OPTIONS_FLAGS="-e name=${PLATFORM_CLUSTER_ID} -e domain=${PLATFORM_CLUSTER_ID}.lab.flanksource.com -vv"
+
+export PLATFORM_CONFIG=$PWD/test/vsphere/e2e.yml
```

Comment:
```suggestion
export PLATFORM_CONFIG=$PWD/test/vsphere/e2e.yml,$PWD/test/common.yml
```

We should probably piggy back of the kind config file and only override the vsphere specifc items

Reply:
Yeah I agree. let's rely on kind common YAML and overwrite only what is needed.

Reply:
I had to revert piggyback here. It does not work properly. When used the manifest as result is:

```yaml
kubernetes:
  version: v1.15.7
  kubeletExtraArgs:
    node-labels: "ingress-ready=true"
    authorization-mode: "AlwaysAllow"
  masterIP: localhost
```

which lead the script to fail because master cannot be found.